src/main/java/com/couchbase/lite/router/Router.java

```diff
@@ -1264,6 +1262,7 @@ public Status do_POST_Document_revs_diff(Database _db, String _docID, String _at
 
         // Look them up, removing the existing ones from revs:
         try {
+            // query db only, not necessary to be syncrhonized
```

Comment:
As [findMissingRevisions()](https://github.com/couchbase/couchbase-lite-java-core/blob/a35361bc47ecb6c55849694deece22fa4aaf7c62/src/main/java/com/couchbase/lite/store/SQLiteStore.java#L963) does not access shard instance and SQLiteDatabae is thread-safe as default, I believe this method is thread-safe.